(abandoned) Move labels to ep

amreshakim opened the pull request:
Decouple labels from containers and store them in endpoints.
- change api to allow passing labels during endpoint creation
- regenerate swagger API
- remove labels from containers
- store labels in endpoints

https://github.com/cilium/cilium/issues/1041

aalemayhu committed:
byteorder: add documentation file for byteorder package

this should show a synopsis on godoc[0].

[0]: https://godoc.org/github.com/cilium/cilium

Closes: #285 (Micro: Fix common.Swab16 with proper encoding/binary usage)
Signed-off-by: Alexander Alemayhu <alexander@alemayhu.com>

aalemayhu committed:
tests: check printed container ids

Related-to: #746 (container id in BPF debug message is wrong)
Signed-off-by: Alexander Alemayhu <alexander@alemayhu.com>

aalemayhu committed:
bpf: don't call bpf_htonl for LXC_IPV4

It's should already be in network byte order from Go side.

Related-to: #285 (Micro: Fix `common.Swab16` with proper `encoding/binary` usage)
Signed-off-by: Alexander Alemayhu <alexander@alemayhu.com>

aalemayhu committed:
pkg: introduce new package for handling byteorder

netlink is used for the byte order detection, which is already a
dependency available in the vendor directory. The package provides the
native byte order in the variable Native and wraps all byte order
specific reads and write call sites.

Related-to: #285 (Micro: Fix `common.Swab16` with proper `encoding/binary` usage)
Signed-off-by: Alexander Alemayhu <alexander@alemayhu.com>

amreshakim committed:
cilium-cni: move labels to ep

- decouple labels from containers
- use endpoint labels

amreshakim committed:
cilium-cni: move labels to ep

- decouple labels from containers
- use endpoint labels

Commit:
change api

allow passing labels while creating an endpoint

amreshakim committed:
regenerate cilium API

tgraf commented on daemon/docker_watcher.go:
```diff
@@ -238,13 +238,8 @@ func (d *Daemon) handleCreateContainer(id string, retry bool) {
-					endpointmanager.LinkContainerID(ep)
-				}
+				// Associate container id with endpoint
+				endpointmanager.LinkContainerID(ep)
```
Calling `LinkContainerID` only makes sense after setting `ep.DockerID`, in this branch, we know that the endpoint exists (it was retrievable by endpoint ID but not by DockerID). I think we should keep the code to set `ep.DockerID`

tgraf commented on daemon/docker_watcher.go:
```diff
@@ -258,35 +253,31 @@ func (d *Daemon) handleCreateContainer(id string, retry bool) {
 		}
 
+		ep.Mutex.Lock()
+		ep.DockerID = id
```
Or we can call `endpointmanager.LinkContainerID(ep)` here.

amreshakim replied:
good idea!

amreshakim committed:
tgraf's comment

dqminh committed:
move from godeps to https://github.com/LK4D4/vndr

Godeps make it very difficult to update only a few dependencies. We uses
vndr in a couple of projects and it works well so far in terms of ease
of dependencies' update.

One more plus is vndr also removes more files that are unnecessary for
the build process.

Signed-off-by: Daniel Dao <dqminh89@gmail.com>

eloycoto committed:
[BPF/probes]: Added kernel checks Fixes: #891

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

amreshakim committed:
update labels on ep creation if needed

tgraf committed:
k8s: Correctly implement ClusterIP for type NodePort and LoadBalancer

NodePort and LoadBalancer are supersets of ClusterIP. Implement
ClusterIP for these types as well.

Also: Silently ingore external name services instead of printing a
warning.

Signed-off-by: Thomas Graf <thomas@cilium.io>

tgraf committed:
examples/demo: Clean up star wars demo

Signed-off-by: Thomas Graf <thomas@cilium.io>

amreshakim committed:
small changes

dqminh committed:
vendor.conf: update etcd to 3.2.1

The move to vendor.conf missed a separate bump for etcd in another change. This
readds the bump.

Signed-off-by: Daniel Dao <dqminh89@gmail.com>

dqminh committed:
remove unused dependencies detected by vndr

Looks like we dont use any of the current prometheus dependencies. vndr detects
those are unused and remove them. `make` still works after the purge.

The original change adds prometheus together with the bump to etcd v3.1, but we
dont actually exposes the prometheus metrics anywhere.

I think we should just remove this now and re-integrate prometheus separately
when working on https://github.com/cilium/cilium/issues/165 maybe.

Signed-off-by: Daniel Dao <dqminh89@gmail.com>

dqminh committed:
remove Godeps manifest since we moved to vndr

Commit 71fefcaf moves the dependencies management to vndr, but forgot to remove
existing Godeps manifest.

Signed-off-by: Daniel Dao <dqminh89@gmail.com>

aanm committed:
tests: adding l7 stresstest on different namespaces

Add L7 stress test on a single node for the L7 proxy.

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
K8s: keeping namespace label set in cilium NP

Cilium Network Policy only enforces policies on the namespace where it's
stored. This is achieved by adding a special label on the
EndpointSelector, on the FromEndpoints and on the FromRequires with the
k8s namespace where the policy is stored.

Since users are allowed to specify that special label on the
FromEndpoints and / or FromRequires, we should keep the value's label
unchanged.

The special label in the EndpointSelector is only possible to be
overwritten with the metadata's namespace. If the user specifies it
manually, by inserting the special label, a warning is printed in cilium
logs and the label is replaced with the namespace where the policy is
stored.

For example, the following CNP
```
metadata:
  name: "l7-stresstest"
  namespace: "development"
spec:
  endpointSelector:
    matchLabels:
      id: server
  ingress:
  - fromEndpoints:
    - matchLabels:
        id: client
        "k8s:io.kubernetes.pod.namespace": qa
```

Will be translated into:
```
"endpointSelector": {
  "matchLabels": {
    "any:id": "server",
    "k8s:io.kubernetes.pod.namespace": "development"
  }
},
"ingress": [
  {
    "fromEndpoints": [
      {
        "matchLabels": {
          "any:id": "client",
          "k8s:io.kubernetes.pod.namespace": "qa"
        }
      }
    ],
```

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
tests: ignoring .json files in 00-no-sleep.sh

Since 00-no-sleep.sh was checking for all files, the commit b7bec4bfd7
add a sleep in the pod's command file causing the script to give a
false positive. This commit ignores all files with the name ".json" to
prevent this false positive.

Fixes: b7bec4bfd7 ("tests: adding l7 stresstest on different namespaces")

Signed-off-by: André Martins <andre@cilium.io>

tklauser committed:
pkg/bpf: add new sock ops prog type

Add prog type ProgTypeSockOps to match BPF_PROG_TYPE_SOCK_OPS,
introduced in the net-next tree in commit 40304b2a1567 ("bpf: BPF
support for sock_ops").

Signed-off-by: Tobias Klauser <tklauser@distanz.ch>

aanm committed:
Jenkinsfile: add timestamps to builds

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
endpointmanager: fixed remove locked for DockerEndpointID

The endpointsAux stores the same EP value with different keys. When
removing the endpoint, all of those keys where removed from the map
except the ep.DockerEndpointID since it was wrongly set with
ep.DockerID

Signed-off-by: André Martins <andre@cilium.io>
